### PR TITLE
Make nav entries dynamic and add admin link

### PR DESCRIPTION
**Pull request by Leppunen**
Menu entries are now dynamically shown and hidden based on login state.
Also a link to the admin page is now added to navbar

### CODE REVIEW

File: web/src/js/Menu.jsx [hunk 1]
```diff
@@ -30,7 +38,7 @@ export default class Menu extends Component {
         <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
           <div className="navbar-nav">
 {this.menuItems.map((menuItem, index) =>
-            <a key={index} className={"nav-item nav-link "+(window.location.pathname == menuItem.link ? "active" : "")} href={menuItem.link}>{menuItem.name}</a>
+            <a key={index} className={`nav-item nav-link ${window.location.pathname == menuItem.link ? "active" : ""} ${menuItem.requireLogin && !isLoggedIn() ? 'd-none' : ''}`} href={menuItem.link}>{menuItem.name}</a>
```
Comment (RAnders00) [inline web/src/js/Menu.jsx:41]: Would appreciate if you could do this with a `.filter()` call before the `.map()`

**Review by RAnders00 — CHANGES_REQUESTED** [review]

File: web/src/js/Menu.jsx (hunk 1 above)
Comment (Leppunen) [reply web/src/js/Menu.jsx]: Done

File: web/src/js/Menu.jsx [hunk 2]
```diff
@@ -37,9 +38,9 @@ export default class Menu extends Component {
         </button>
         <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
           <div className="navbar-nav">
-{this.menuItems.map((menuItem, index) =>
-            <a key={index} className={`nav-item nav-link ${window.location.pathname == menuItem.link ? "active" : ""} ${menuItem.requireLogin && !isLoggedIn() ? 'd-none' : ''}`} href={menuItem.link}>{menuItem.name}</a>
-          )}
+            {this.menuItems.filter(item => !item.requireLogin || item.requireLogin === isLoggedIn()).map((menuItem, index) =>
```
Comment (RAnders00) [inline web/src/js/Menu.jsx:41]: suggestion: replace `===` with `&&` to clarify what's going on

```suggestion
            {this.menuItems.filter(item => !item.requireLogin || (item.requireLogin && isLoggedIn())).map((menuItem, index) => 
```

**Review by RAnders00 — CHANGES_REQUESTED** [review]

**Comment by alazymeme** [comment]
@pajlada can you review pls image

**Review by pajlada — APPROVED** [review]

**Pull request merged**